src/state/appUiConfigs.ts

```diff
@@ -94,6 +96,18 @@ export const appUiConfigsSlice = createSlice({
         })
       );
     },
+    favoritedMarket: (state: AppUIConfigsState, { payload: marketId }: PayloadAction<string>) => {
+      const currentFavoritedMarkets = state.favoritedMarkets;
+      const newFavoritedMarkets = [...currentFavoritedMarkets, marketId];
+      state.favoritedMarkets = newFavoritedMarkets;
+      track(AnalyticsEvents.FavoritedMarket({ marketId }));
```

Comment:
(this is fine for now, I know we do it all over) I don't actually think this pattern is super safe. Reducers are meant to never have side effects and might be subject to e.g. replays and other weirdness.

Reply:
Moved this out to the components onToggle